src/components/VoteBox.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import styled from "styled-components";
+export default function VoteBox() {
+  const [voteDataList, setVoteDataList] = useState([]);
+
+  const sortVoteDataList = (dataList) => {
+    const returnDataList = dataList
+      .concat()
+      .sort((a, b) => (a.voteCount > b.voteCount ? -1 : 1));
+    return returnDataList;
+  };
+
+  useEffect(() => {
+    const fetchVoteData = async () => {
+      try {
+        const response = await axios.get(
+          "http://ec2-3-34-5-220.ap-northeast-2.compute.amazonaws.com:2020/candidates"
+        );
+        const initialVoteDataList = sortVoteDataList(response.data);
+        setVoteDataList(initialVoteDataList);
+      } catch (e) {
+        console.log(e);
+      }
+    };
+    fetchVoteData();
+  }, []);
+
+  const sendVoteResult = async (id, name) => {
+    try {
+      const response = await axios.get(
+        `http://ec2-3-34-5-220.ap-northeast-2.compute.amazonaws.com:2020/vote?id=${id}`
+      );
+      //아 get으로도 보내는건가..? post를 안써도 되는거야?
+      if (response) {
+        alert(`${name}님께 투표 완료!`);
+      }
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  const handleVoteButton = (id, name) => {
+    sendVoteResult(id, name);
+    const nextVoteDataList = voteDataList.map((data, index) => {
+      if (data.id !== id) return data;
+      return {
+        ...data,
+        voteCount: data.voteCount + 1,
+      };
+    });
+
+    const sortedVoteDataList = sortVoteDataList(nextVoteDataList);
+
+    setVoteDataList(sortedVoteDataList);
+  };
+
+  const newVoteDataList = voteDataList.map((data, index) => (
+    <VoteDataWrapper>
+      <VoteDataList key={data.id}>
+        {index + 1}위: {data.name} [{data.voteCount}표]{" "}
+      </VoteDataList>
+      <VoteButton onClick={() => handleVoteButton(data.id, data.name)}>
+        투표
+      </VoteButton>
+    </VoteDataWrapper>
+  ));
```

Comment:
1. 계속 `data.`으로 가져오기 귀찮으니까 이렇게 한번에 뽑아서 사용해도 좋을 것 같아요~

2. `handleVoteCount` 네이밍 변경 확인해주세요~

3. `map`할때 부여하는 `key`는 최상위 컴포넌트에 넣어주세요 : )


```suggestion
  const newVoteDataList = voteDataList.map((data, index) =>{ 
    const {id,voteCount,name}=data;
    return(
    <VoteDataWrapper key={id}>
      <VoteDataList >
        {index + 1}위: {name} [{voteCount}표]
      </VoteDataList>
      <VoteButton onClick={() => handleVoteCount(id, name)}>
        투표
      </VoteButton>
    </VoteDataWrapper>
  )
});
```

Reply:
arrow function을 하면 바로 return 값을 줘야 하는 줄 알았는데.. { } 를 치면 코드를 추가할 수 있군요!!
그리고 destructing을 해주니 확실히 코드가 더 깔끔해 보입니당!! map function key에 대해서는 더 공부해보겠습니다!!
🙇